packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -956,6 +991,24 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
       );
     }
 
+    if (widget.endDrawer != null) {
+      assert(hasEndDrawer);
+      _addIfNonNull(
+        children,
+        new DrawerController(
+          key: _endDrawerKey,
+          type: DrawerAlignment.END,
+          child: widget.endDrawer,
+        ),
+        _ScaffoldSlot.endDrawer,
+        // remove the side padding from the side we're not touching
+        removeLeftPadding: textDirection == TextDirection.rtl,
```

Comment:
I would have expected these to flip around compared to the `DrawerAlignment.start` version.

Reply:
Fixed